bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java

```diff
@@ -203,6 +204,9 @@ private void registerForLifecycleEvents(@NonNull Context context) {
                 new ActivityLifecycleCallbacksAdapter() {
                     @Override
                     public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+                        if (openedFromDeeplink(activity)) {
+                            return;
+                        }
```

Comment:
So I think if we roughly keep this solution in some form, this check would at least need to be moved to _after_ the setting of the `mIsClearAllowed` / `mIsConfigChange` variables:

```java
@Override
public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
    mIsClearAllowed = true;
    mIsConfigChange = false;

    if (openedFromDeeplink(activity)) {
       return;
    }

    ...
}
```

Reply:
Hmmm, correct, I was thinking that could do clear later on if user open other activity, but what if there are some activity stored behind and new activity will get cache removed, then back to those stored activity and restore will fail.

Reply:
Well the `mIsClearAllowed` and `mIsConfigChange` values just control some internal behavior and don't affect overall clearing of all cached data. The `mIsClearAllowed` is just a check to make sure the `Bridge.clear(this)` calls don't actually do anything during certain scenarios (like configuration changes and when Don't Keep Activities is enabled). The `mIsConfigChange` controls whether data processing occurs on the main thread or on a background thread. So we just want to make sure these values are still set as normal, since all your code is trying to do is prevent the overall data clearing from occurring and that's strictly related to all the code below.